schema/embed.go

```diff
@@ -0,0 +1,57 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package schema
+
+import (
+	"embed"
+	"io/fs"
+)
+
+//go:embed *
+var assets embed.FS
+
+// Assets returns a file system with the contents of the schema directory
+func Assets() fs.FS {
+	return assets
+}
+
+// Paths returns a list of paths to directories within the schema subdirectory that have versioned schemas in them
+func Paths(dbSubDir string) []string {
+	efs := Assets()
+	dirs := make([]string, 0)
+	_ = fs.WalkDir(efs, dbSubDir, func(path string, d fs.DirEntry, err error) error {
```

Comment:
You should return this error from the function as well so we don't fail silently

Reply:
done